### PR TITLE
Also compute relative error

### PR DESCRIPTION
**Pull request by sebwolf-de**
Hi, I'd like to also compute relative errors, when doing a convergence analysis.

### CODE REVIEW

File: src/ResultWriter/AnalysisWriter.cpp [hunk 1]
```diff
@@ -200,11 +225,11 @@ void seissol::writer::AnalysisWriter::printAnalysis(double simulationTime) {
         } else {
           MPI_Recv(centerRecv, 3, MPI_DOUBLE,  errLInfRecv[i].rank, i, comm, MPI_STATUS_IGNORE);
         }
-        logInfo(mpi.rank()) << "L1  , var[" << i << "] =\t" << errL1MPI[i];
-        logInfo(mpi.rank()) << "L2  , var[" << i << "] =\t" << std::sqrt(errL2MPI[i]);
-        logInfo(mpi.rank()) << "LInf, var[" << i << "] =\t" << errLInfRecv[i].val
+        logInfo(mpi.rank()) << "L1  , var[" << i << "] =\t" << errL1MPI[i] << "	" << errL1MPI[i] / analyticalL1MPI[i];
```
Comment (krenzland) [inline src/ResultWriter/AnalysisWriter.cpp:228]: Why are you using this kind of spacing? Why no tabs? Would be also good use the same spacing everywhere?

**Comment by Thomas-Ulrich** [comment]
Could you add in the documentation how to run such convergence analysis?
Is it related to this cube setup with periodic boundaries?

File: src/ResultWriter/AnalysisWriter.cpp [hunk 2]
```diff
@@ -215,10 +240,10 @@ void seissol::writer::AnalysisWriter::printAnalysis(double simulationTime) {
...
             center);
 
-      logInfo() << "L1, var[" << i << "] =\t" << errL1Local[i];
-      logInfo() << "L2, var[" << i << "] =\t" << std::sqrt(errL2Local[i]);
-      logInfo() << "LInf, var[" << i << "] =\t" << errLInfLocal[i]
-          << "\tat [" << center[0] << ",\t" << center[1] << ",\t" << center[2] << "\t]";
+      logInfo() << "L1, var[" << i << "] =\t" << errL1Local[i] << "\t" << analyticalL1Local[i];
+      logInfo() << "L2, var[" << i << "] =\t" << std::sqrt(errL2Local[i]) << "\t" << std::sqrt(analyticalL2Local[i]);
+      logInfo() << "LInf, var[" << i << "] =\t" << errLInfLocal[i << "\t" << analyticalLInfLocal[i]]
+          << "\tat [" << center[0] << "	" << center[1] << "	" << center[2] << "\t]"
```
Comment (Thomas-Ulrich) [inline src/ResultWriter/AnalysisWriter.cpp:246]: not fixed tab issue here

**Comment by krenzland** [comment]
> Could you add in the documentation how to run such convergence analysis?
> Is it related to this cube setup with periodic boundaries?

This is done automatically when running a simulation with certain initial conditions.
I don't think that it is documented currently but it's not in scope of this PR.
So I'd make this a separate issue - normal users won't be interested in convergence analysis anyway.

**Comment by Thomas-Ulrich** [comment]
> > Could you add in the documentation how to run such convergence analysis?
> > Is it related to this cube setup with periodic boundaries?
> 
> This is done automatically when running a simulation with certain initial conditions.
> I don't think that it is documented currently but it's not in scope of this PR.
> So I'd make this a separate issue - normal users won't be interested in convergence analysis anyway.

Everything you put in the documentation is something you do not need to give an answer to later one, so the effort is not necessarily unworthy. But in the end, it is up to you.

**Comment by krenzland** [comment]
No, I agree that we should add this. I just want to avoid the situation where a lot of extra work is attached to every PR. We want to merge stuff as fast as possible.

When you modify something that is documented, I agree that we should modify the documentation in the same PR.

File: src/ResultWriter/AnalysisWriter.cpp (hunk 2 above)
Comment (sebwolf-de) [reply src/ResultWriter/AnalysisWriter.cpp]: Fixed, thanks for pointing this out!

**Comment by sebwolf-de** [comment]
There is a PR in the pipeline connected to https://github.com/SeisSol/Examples/issues/10. Once the licence debate is over, I plan to merge a script (originally written by Carsten) to compute convergence rates and so forth. This will include docu.

**Comment by Thomas-Ulrich** [comment]
> No, I agree that we should add this. I just want to avoid the situation where a lot of extra work is attached to every PR. We want to merge stuff as fast as possible.
> 
> When you modify something that is documented, I agree that we should modify the documentation in the same PR.

Ok I understand, but 
Is it a lot of extra work? Probably this won't exceed 10 lines of doc.
It is clear that opening an issue has little chance to lead to a documentation.

**Comment by krenzland** [comment]
> Is it a lot of extra work? Probably this won't exceed 10 lines of doc.
> It is clear that opening an issue has little chance to lead to a documentation.

No, not really. (Well, it's still ~10 minutes to write and actually test the documentation).
 But it's something that I noticed for a few past PRs that a lot of additional changes get tacked on top of them. PRs should be (imo) as small as possible to ensure that we can merge it.

But I generally agree. Maybe the best idea is to write documentation when you need to explain something. I've explained the convergence procedure at least once and haven't written the documentation...

**Pull request merged**